### PR TITLE
Improve upload artifact to builds.hex.pm workflow

### PR DESCRIPTION
**Pull request by wingyplus**
In this changeset, I changed the upload workflow to upload after we built the release artifact. So the flow will be changed to: 

Screenshot 2024-09-20 235624

You can see the demo at https://github.com/wingyplus/elixir/actions/runs/11056313905.

The pros of this way is we have only a single workflow, can reuse all both nightly (main) and release (v*) versions. 

I have a few tasks left to unlock the draft:

- [x] Uncomment Fastly flush cache. I cannot test this step by myself because I have no Fastly setup. :(
- [x] Make update builds.txt works.
- [x] Remove builds.hex.pm.yml

Before continue those tasks above, is this PR ok for you?

### CODE REVIEW

**Comment by wojtekmach** [comment]
@wingyplus yup, the direction looks good to me!

**Comment by wojtekmach** [comment]
> Uncomment Fastly flush cache. I cannot test this step by myself because I have no Fastly setup. :(

If it's easy enough to move that into a separate step, perhaps guarding it like the following will help you?

```yaml
name: Flush Fastly cache
if: github.repository == 'elixir-lang/elixir'
run: |
  ...
```

**Comment by wingyplus** [comment]
@wojtekmach The PR is ready to review now. 🙏 



> > Uncomment Fastly flush cache. I cannot test this step by myself because I have no Fastly setup. :(
> 
> If it's easy enough to move that into a separate step, perhaps guarding it like the following will help you?
> 
> ```yaml
> name: Flush Fastly cache
> if: github.repository == 'elixir-lang/elixir'
> run: |
>   ...
> ```

That's a good idea. I will try it tomorrow. 👍

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -149,3 +149,112 @@ jobs:
...
+        run: |
+          cat << 'EOF' > utils.sh
+          function purge_key() {
+            curl \
+               -X POST \
+               -H "Fastly-Key: ${FASTLY_KEY}" \
+               -H "Accept: application/json" \
+               -H "Content-Length: 0" \
+               "https://api.fastly.com/service/$1/purge/$2"
+            echo 'Temporary close'
```
Comment (wingyplus) [inline .github/workflows/release.yml:179]:
```suggestion
```

**Comment by wingyplus** [comment]
> @wojtekmach The PR is ready to review now. 🙏
> 
> > > Uncomment Fastly flush cache. I cannot test this step by myself because I have no Fastly setup. :(
> > 
> > 
> > If it's easy enough to move that into a separate step, perhaps guarding it like the following will help you?
> > ```yaml
> > name: Flush Fastly cache
> > if: github.repository == 'elixir-lang/elixir'
> > run: |
> >   ...
> > ```
> 
> That's a good idea. I will try it tomorrow. 👍

Complete! Here is the demo https://github.com/wingyplus/elixir/actions/runs/11056148036/job/30717118796. The flush cache disabled because of `if` guard.

**Review by wojtekmach — APPROVED** [review]
Looks good to me!

**Review by ericmj — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**